apollo-federation/src/sources/connect/json_selection/apply_to.rs

```diff
@@ -69,6 +73,44 @@ impl JSONSelection {
 
         (value, errors.into_iter().collect())
     }
+
+    pub fn shape(&self) -> Shape {
```

Comment:
what's the reason for having shape code in this file? it feels different enough to warrant a separate file and maybe a separate trait, even if it's structured basically the same

Reply:
It made the code a little easier to write to have `apply_to_path` nearby, but there's no harm in moving it to a new file now.